### PR TITLE
Frontend proto

### PR DESCRIPTION
**Pull request by sofiatengvall**

### CODE REVIEW

File: client/src/components/floating-panel-view/floating-panel-view.scss [hunk 1]
```diff
@@ -0,0 +1,43 @@
+@import '../../scss/atoms/button';
```
Comment (HenrikHelenius) [inline client/src/components/floating-panel-view/floating-panel-view.scss]: Order correctly: vars, cols, mixins, button

File: client/src/components/floating-panel-view/floating-panel-view.scss [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+.panel {
+  width: 40vh;
+  height: 90vh;
+  position: fixed;
+  display: flex;
+  background-color: $palette-light;
+  border: solid $palette-medium 1px;
+  box-shadow: 2px 2px 5px $palette-medium;
```
Comment (HenrikHelenius) [inline client/src/components/floating-panel-view/floating-panel-view.scss]: design comment, don't make it slide in x axis, so e.g. 0 2px 10px, and try lighter shadow than the border is, otherwise it will just look like a fuzzy border

File: client/src/components/floating-panel-view/floating-panel-view.scss [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+.panel-header {
+  display: flex;
+  justify-content: flex-end;
+  height: 5%;
+  width: 100%;
+  background-color: $palette-medium;
+  border-bottom: solid $palette-medium 1px;
```
Comment (HenrikHelenius) [inline client/src/components/floating-panel-view/floating-panel-view.scss]: border bottom after width, then bg-color

File: client/src/components/open-panel-button/open-panel-button.scss [hunk 4]
```diff
@@ -0,0 +1,28 @@
+@import '../../scss/atoms/button';
+@import '../../scss/mixins';
+@import '../../scss/variables';
+@import '../../scss/colors';
+
+.button {
+  @include button(30px);
```
Comment (HenrikHelenius) [inline client/src/components/open-panel-button/open-panel-button.scss]: use relative like 100% instead

**Review by HenrikHelenius — CHANGES_REQUESTED** [review]

File: client/src/components/open-panel-button/open-panel-button.scss [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  @include display-and-animate();
+  position: fixed;
+  width: 30px;
+  height: 30px;
+  bottom: 5%;
+  left: 3%;
+  background: $palette-accent;
+  box-shadow: 2px 2px 5px $palette-dark;
```
Comment (HenrikHelenius) [inline client/src/components/open-panel-button/open-panel-button.scss]: use the new shadow here, why x axis sliding?

File: client/src/scss/_colors.scss [hunk 6]
```diff
@@ -0,0 +1,4 @@
+$palette-light: #FDFDFD;
```
Comment (HenrikHelenius) [inline client/src/scss/_colors.scss]: all to lowercase in hex

File: client/src/scss/_mixins.scss [hunk 7]
```diff
@@ -0,0 +1,11 @@
+@mixin hide-and-animate($duration: $default-animation-duration) {
+  visibility: hidden;
+  opacity: 0;
+  transition: visibility $duration linear 200ms, opacity 200ms;
```
Comment (HenrikHelenius) [inline client/src/scss/_mixins.scss]: 200ms = 0.2s

File: client/src/scss/_variables.scss [hunk 8]
```diff
@@ -0,0 +1,8 @@
+/* transitions */
+$default-animation-duration: 0.2ms;
```
Comment (HenrikHelenius) [inline client/src/scss/_variables.scss]: 0.2ms != 200ms

File: client/src/scss/atoms/_button.scss [hunk 9]
```diff
@@ -0,0 +1,7 @@
+@mixin button($border: $default-border-radius) {
+  /*reset*/
+  padding: 2px;
+  border-radius: $border;
```
Comment (HenrikHelenius) [inline client/src/scss/atoms/_button.scss]: $border?

File: client/src/scss/atoms/_button.scss [hunk 10]
```diff
@@ -0,0 +1,7 @@
...
+  border: none;
```
Comment (HenrikHelenius) [inline client/src/scss/atoms/_button.scss:3]: first border then border radius

File: client/src/scss/_base.scss [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+}
+
+// Ensure that feedbacker tool is always on top
+.feedback-app-main-container {
+  position: absolute;
+  top: 0;
+  left: 0;
+  height: 100%;
+  width: 100%;
+  z-index: 100000;
```
Comment (HenrikHelenius) [inline client/src/scss/_base.scss]: make z-index plan somewhere e.g. in variables, extract this to var

File: client/src/index.scss [hunk 12]
```diff
@@ -0,0 +1,24 @@
+@import 'colors';
```
Comment (HenrikHelenius) [inline client/src/index.scss]: remove file

**Review by HenrikHelenius — CHANGES_REQUESTED** [review]

File: client/src/components/floating-panel-view/floating-panel-view.scss [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+.panel {
+  width: 40vh;
+  height: 90vh;
+  position: fixed;
+  display: flex;
+  background-color: $palette-light;
+  border: solid $palette-medium 1px;
+  box-shadow: 0px 2px 10px $palette-shadow-default;
```
Comment (HenrikHelenius) [inline client/src/components/floating-panel-view/floating-panel-view.scss]: 0px unnecessary change to 0

File: client/src/components/open-panel-button/open-panel-button.scss [hunk 14]
```diff
@@ -0,0 +1,28 @@
+@import '../../scss/variables';
+@import '../../scss/colors';
+@import '../../scss/mixins';
+@import '../../scss/atoms/button';
+
+.button {
+  @include button(100%);
+  @include display-and-animate();
+  position: fixed;
+  width: 30px;
```
Comment (HenrikHelenius) [inline client/src/components/open-panel-button/open-panel-button.scss]: make local variable in scope

File: client/src/components/open-panel-button/open-panel-button.scss [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+  height: 30px;
+  bottom: 5%;
+  left: 3%;
+  background: $palette-accent;
+  box-shadow: 0px 2px 5px $palette-shadow-default;
```
Comment (HenrikHelenius) [inline client/src/components/open-panel-button/open-panel-button.scss]: remove px

File: client/src/scss/_mixins.scss [hunk 16]
```diff
@@ -0,0 +1,11 @@
...
+}
+
+@mixin display-and-animate($duration: $default-animation-duration) {
+  visibility: visible;
+  opacity: 1;
+  transition: visibility $duration linear 200ms, opacity 200ms;
```
Comment (HenrikHelenius) [inline client/src/scss/_mixins.scss]: duration not used

File: client/src/components/floating-panel-view/floating-panel-view.scss [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+  justify-content: flex-end;
+  height: 5%;
+  width: 100%;
+  background-color: $palette-medium;
+  cursor: grab;
+}
+
+.panel-header button {
+  @include button;
+  height: 70%;
```
Comment (HenrikHelenius) [inline client/src/components/floating-panel-view/floating-panel-view.scss]: what is this?

File: client/src/assets/svg/baseline-add-24px.svg [hunk 18]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"/><path d="M0 0h24v24H0z" fill="none"/></svg>
```
Comment (bqqbarbhg) [inline client/src/assets/svg/baseline-add-24px.svg:1]: Some license stuff to mention (?)

**Review by bqqbarbhg — CHANGES_REQUESTED** [review]

File: client/src/assets/svg/baseline-add-24px.svg (hunk 18 above)
Comment (HenrikHelenius) [reply client/src/assets/svg/baseline-add-24px.svg:1]: in notice

File: client/src/components/floating-panel-view/floating-panel-view.scss [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+
+.panel-container {
+  @include display-and-animate($default-animation-duration);
+  &.hidden {
+    @include hide-and-animate($default-animation-duration);
+  }
+}
+
+.panel {
+  width: 40vh;
```
Comment (HenrikHelenius) [inline client/src/components/floating-panel-view/floating-panel-view.scss]: w: 40vw,
min-width: 200px;

**Review by HenrikHelenius — DISMISSED** [review]

File: client/src/components/floating-panel-view/floating-panel-view.scss [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+  height: 90vh;
+  position: fixed;
```
Comment (HenrikHelenius) [inline client/src/components/floating-panel-view/floating-panel-view.scss:14]: position fixed first in .panel

File: client/src/components/floating-panel-view/floating-panel-view.scss [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+  display: flex;
```
Comment (HenrikHelenius) [inline client/src/components/floating-panel-view/floating-panel-view.scss:15]: display before widths

File: client/src/components/floating-panel-view/floating-panel-view.scss [hunk 22]
```diff
@@ -0,0 +1,41 @@
...
+  background-color: $palette-light;
```
Comment (HenrikHelenius) [inline client/src/components/floating-panel-view/floating-panel-view.scss]: background last

File: client/src/components/floating-panel-view/floating-panel-view.scss [hunk 23]
```diff
@@ -0,0 +1,41 @@
...
+.panel-header {
+  display: flex;
+  justify-content: flex-end;
+  height: 5%;
+  width: 100%;
+  background-color: $palette-medium;
+  cursor: grab;
+}
+
+.panel-header button {
```
Comment (HenrikHelenius) [inline client/src/components/floating-panel-view/floating-panel-view.scss]: move panel-header button to .panel-header { button ... }

**Comment by bqqbarbhg** [comment]
Cross-browser tests:

- IE11
  - Looks pretty good
  - Closing the box is a little delayed
  - The hover cursor doesn't work
- Firefox
  - Works as well as Chrome
  - Grabbing cursor is a pixelated mess.

File: client/src/scss/atoms/_button.scss [hunk 24]
```diff
@@ -0,0 +1,7 @@
+@mixin button($border-radius: $default-border-radius) {
+  /*reset*/
+  padding: 2px;
```
Comment (HenrikHelenius) [inline client/src/scss/atoms/_button.scss]: padding 2px to another place

**Comment by HenrikHelenius** [comment]
Tested safari Version 12.0.1 (14606.2.104.1.1)

**Comment by bqqbarbhg** [comment]
Works on IE11 and Firefox

**Review by bqqbarbhg — DISMISSED** [review]

**Review by HenrikHelenius — DISMISSED** [review]

**Comment by HenrikHelenius** [comment]
[(0/5) [story] [frontend proto] React renders a floating panel Sofia](https://trello.com/c/Mdd7EN0u/61-0-5-story-frontend-proto-react-renders-a-floating-panel-sofia)

**Comment by HenrikHelenius** [comment]
[(0/5) [story] [frontend proto] React renders a floating panel Sofia](https://trello.com/c/Mdd7EN0u/61-0-5-story-frontend-proto-react-renders-a-floating-panel-sofia)

**Comment by HenrikHelenius** [comment]
[(0/5) [story] [frontend proto] SASS styling for React Henrik](https://trello.com/c/uYaZqHJg/65-0-5-story-frontend-proto-sass-styling-for-react-henrik)

**Review by HenrikHelenius — APPROVED** [review]
